src/dlio_benchmark.py

```diff
@@ -232,49 +207,23 @@ def _eval(self, epoch):
         self.args.reconfigure(epoch, DatasetType.VALID)
         step = 1
         total = math.floor(self.num_samples * self.num_files_eval / self.batch_size_eval / self.comm_size)
-
         loader = self.framework.get_loader(DatasetType.VALID)
         loader.read()
-
-        start_time = time()
-        total_compute_time = 0.0
-        time_epoch = {}
-        time_epoch['epoch'] = epoch
-        time_epoch['time_per_step'] = []
         t0 = time()
         for batch in dlp.iter(loader.next()):
-            t1 = time()
             self.stats.eval_batch_loaded(epoch, step, t0)
             if self.eval_time > 0:
                 if self.eval_time_stdev > 0:
                     eval_time = random.normal(self.eval_time, self.eval_time_stdev)
                 else:
                     eval_time = self.eval_time
-                if step > 1:
-                    total_compute_time += eval_time
                 self.framework.compute(epoch, step, eval_time)
-            t1 = time()
-            time_epoch['time_per_step'].append(t1 - t0)
-            self.stats.eval_batch_processed(epoch, step, t0)
+            self.stats.eval_batch_processed(epoch, step, t0, eval_time)
 
             step += 1
             if step > total:
                 break
             t0 = time()
-            self.framework.barrier()
```

Comment:
Did we remove barrier per step?

Reply:
No. Barrier per step is needed. Each time step, allreduce is called in actual training